lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2242,9 +2242,47 @@ struct IntelLSUControlsInfo {
   llvm::Optional<unsigned> PrefetchInfo;
 };
 
-IntelFPGADecorations
-tryParseIntelFPGAAnnotationString(StringRef AnnotatedCode) {
-  IntelFPGADecorations Decorates;
+void processOptionalAnnotationInfo(Constant *Const,
+                                   std::string &AnnotationString) {
+  // Handle optional parameter. It can be for example
+  // { %struct.S, i8*, void ()* } { %struct.S undef, i8* null,
+  //                                void ()* @_Z4blahv }
+  // Now we will just handle integer constants (wrapped in a constant
+  // struct, that is being bitcasted to i8*), converting them to string.
+  // TODO: remove this workaround when/if an extension spec that allows or adds
+  // variadic-arguments UserSemantic decoration
+  if (!Const->getNumOperands())
+    return;
+  if (auto *CStruct = dyn_cast_or_null<ConstantStruct>(Const->getOperand(0))) {
```

Comment:
It seems we expect that `Const` has at least one operand, what is the point to use `dyn_cast_or_null` then?

Reply:
No need for that, changed to dyn_cast